xarray/backends/common.py

```diff
@@ -267,13 +268,23 @@ def robust_getitem(array, key, catch=Exception, max_retries=6, initial_delay=500
             time.sleep(1e-3 * next_delay)
 
 
-class BackendArray(NdimSizeLenMixin, indexing.ExplicitlyIndexed):
+class BackendArray(ABC, NdimSizeLenMixin, indexing.ExplicitlyIndexed):
     __slots__ = ()
 
+    @abstractmethod
+    def __getitem__(key: indexing.ExplicitIndexer) -> np.typing.ArrayLike: ...
+
+    async def async_getitem(key: indexing.ExplicitIndexer) -> np.typing.ArrayLike:
+        raise NotImplementedError("Backend does not not support asynchronous loading")
```

Comment:
I've implemented this for the `ZarrArray` class but in theory it could be supported by other backends too.

Reply:
This might not be the desired behaviour though - this currently means if you opened a dataset from netCDF and called `ds.load_async` you would get a `NotImplementedError`. Would it be better to quietly just block instead?

Reply:
Yes absolutely.

Reply:
Okay I can do that. But can you explain why you feel that this would be better behaviour? Asking for something to be done async and it quietly blocking also seems not great...